Mods/SML/Source/SML/Public/SessionSettings/SessionSettingsManager.h

```diff
@@ -74,6 +74,26 @@ class SML_API USessionSettingsManager : public UWorldSubsystem, public IFGOption
 	bool IsInMainMenu() const;
 	
 	void OnOptionUpdated(FString String, FVariant Value) const;
+
+	/*
+	 * Using (integer) literals for template argument deduction is troublesome
+	 * because the deduced template type is not visible by default. Given that
+	 * requiring callers to cast literals is hideous and error-prone, use some
+	 * funny template shenanigans to prevent template argument deduction using
+	 * the values passed as function arguments. This requires callers of these
+	 * function templates to explicitly specify the type argument T.
+	 */
+	template<typename T>
+	FORCEINLINE T GetOptionValue_Typed(const FString& cvar, typename TEnableIf<true, const T>::Type defaultValue) const
```

Comment:
Consider using `TIdentity<const T>` instead, I recall seeing comments that describe using `TIdentity<>` to prevent template argument deduction.

Reply:
Yeah, that seems like a cleaner way to do it. There's also `TIdentity_T` to skip the `::Type` part. It's also used by TVariant::Set to ensure the correct TVariant type is used.

https://github.com/EpicGames/UnrealEngine/blob/5.2.1-release/Engine/Source/Runtime/Core/Public/Templates/Identity.h